crates/youki/src/main.rs

```diff
@@ -117,24 +76,31 @@ fn main() -> Result<()> {
         nix::unistd::geteuid(),
         std::env::args_os()
     );
-    let root_path = determine_root_path(opts.root)?;
-    let systemd_cgroup = opts.systemd_cgroup;
+    let root_path = determine_root_path(opts.global.root)?;
+    let systemd_cgroup = opts.global.systemd_cgroup;
 
     match opts.subcmd {
-        SubCommand::Create(create) => create.exec(root_path, systemd_cgroup),
-        SubCommand::Start(start) => start.exec(root_path),
-        SubCommand::Run(run) => run.exec(root_path, systemd_cgroup),
-        SubCommand::Exec(exec) => exec.exec(root_path),
-        SubCommand::Kill(kill) => kill.exec(root_path),
-        SubCommand::Delete(delete) => delete.exec(root_path),
-        SubCommand::State(state) => state.exec(root_path),
-        SubCommand::Info(info) => info.exec(),
-        SubCommand::List(list) => list.exec(root_path),
-        SubCommand::Spec(spec) => spec.exec(),
-        SubCommand::Pause(pause) => pause.exec(root_path),
-        SubCommand::Resume(resume) => resume.exec(root_path),
-        SubCommand::Events(events) => events.exec(root_path),
-        SubCommand::Ps(ps) => ps.exec(root_path),
+        SubCommand::Standard(cmd) => match cmd {
+            StandardCmd::Create(create) => {
+                commands::create::create(create, root_path, systemd_cgroup)
+            }
+            StandardCmd::Start(start) => commands::start::start(start, root_path),
+            StandardCmd::Kill(kill) => commands::kill::kill(kill, root_path),
+            StandardCmd::Delete(delete) => commands::delete::delete(delete, root_path),
+            StandardCmd::State(state) => commands::state::state(state, root_path),
+        },
+        SubCommand::Common(cmd) => match cmd {
+            CommonCmd::Events(events) => commands::events::events(events, root_path),
+            CommonCmd::Exec(exec) => commands::exec::exec(exec, root_path),
+            CommonCmd::List(list) => commands::list::list(list, root_path),
+            CommonCmd::Pause(pause) => commands::pause::pause(pause, root_path),
+            CommonCmd::Ps(ps) => commands::ps::ps(ps, root_path),
+            CommonCmd::Resume(resume) => commands::resume::resume(resume, root_path),
+            CommonCmd::Run(run) => commands::run::run(run, root_path, systemd_cgroup),
+            CommonCmd::Spec(spec) => commands::spec_json::spec(spec),
+        },
+
+        SubCommand::Info(info) => commands::info::info(info),
```

Comment:
I think it would make more sense to create a `SubCommand::Ext` and have the info command belong to it, in the same treatment as `Standard` and `Common`, but what do you think?

Reply:
I don't want `Info` to be in the command line crate, because it is (AFAICT) a youki specific extension, whereas the CLI crate is supposed to be only for handling things common to all (or at least most) runtimes.  I could create an `ExtCommand` enum with the `youki` crate if you like, although I don't see a lot of point to it (we can always add it, if youki gets more extension commands in future).